### ISSUE TITLE
Automated metric logging in Lightning Trainer

### ISSUE THREAD

**Issue by baskrahmer** [PyTorchLightning/pytorch-lightning#17909]
### Description & Motivation

As described in [the docs](https://torchmetrics.readthedocs.io/en/stable/pages/lightning.html), you need to define log statements for each of your metrics in your LightningModule to collect this data during training. Personally I have never created any custom logging metric, and always log fairly standard metrics (accuracy, F1, etc.) for training and validation (on an epoch-level or step-interval-leval basis). This introduces the following repetitive code in my model module for both training and validation:
- Definition of the metrics
- Logging to the metrics on step
- Logging to the metrics on epoch

### Pitch

From my perspective, there could be merit in a `metrics` argument of type `Optional[Union[Metric, Iterable[Metric]]]` that is passed to the `Trainer` object and collects these metrics on a per-epoch or per-step-interval basis for both training and validation. This would give the user the possibility to abstract away the the logging of metrics for the most common usecases to inside the trainer class and not left to the user.

cc @borda
